src/clr-angular/forms/checkbox/checkbox-container.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Component} from "@angular/core";
+
+import {DynamicWrapper} from "../../utils/host-wrapping/dynamic-wrapper";
+import {FormControlService} from "../common/form-control.service";
+
+@Component({
+    selector: "clr-checkbox-container",
+    template: `
+        <!-- We want the radio input to be before the label, always -->
+        <ng-content select="[clrCheckbox]"></ng-content>
+        <ng-content></ng-content>
+        <label *ngIf="_dynamic"></label>
+    `,
+    host: {"[class.checkbox]": "true"},
+    providers: [FormControlService]
+})
+export class ClrCheckboxContainer implements DynamicWrapper {
+    // Indicates whether the container is dynamically created by the radio button itself
+    _dynamic = false;
```

Comment:
This has to be public because another class will set the value, but it's internal wiring so we don't want consumers to mess with this property. This case is addressed in our guidelines:
>Finally, some properties or methods needs to be public for wiring purposes, but we don't want consumers to play with them. These will stay public but be prefixed with an underscore.